src/celeritas/global/CoreTrackData.hh

```diff
@@ -174,7 +174,14 @@ inline void resize(CoreStateData<Ownership::value, M>* state,
     resize(&state->rng, params.rng, size);
     resize(&state->sim, size);
     resize(&state->init, params.init, size);
-    resize(&state->track_slots, size);
+
+    Collection<TrackSlotId, Ownership::value, MemSpace::host, ThreadId> track_slots;
```

Comment:
Initialize the data on the host and then use `Collection` assignation operator to manage the memory.

Using `std::iota`, we'd have to implement `OpaqueID::operator++()` which is immutable so we'd lose this property. For device code, thrust also needs a few different operations on `T` as it's expecting numbers:

```c++
OpaqueId<V, S> operator*(OpaqueId<V, S> id, std::make_signed_t<S> factor);
OpaqueId<V, S> operator+(OpaqueId<V, S> self, OpaqueId<V, S> other);
```